src/components/AuthContainer.ts

```diff
@@ -15,11 +15,11 @@ import TermsOfService from 'src/pages/TermsOfService';
 const AuthContainer = ({ children }) => {
   const { name, public: isPublic } = useRoute();
   const { isSignedIn, registrationStatus, termsOfService, profile } = useStore(authStore);
-  const displayTosPage = isSignedIn && termsOfService.permitsSystemUsage === false;
+  const displayTosPage = isSignedIn === true && (termsOfService as any).permitsSystemUsage === false;
   const seenAzurePreview = useStore(azurePreviewStore) || false;
-  const authspinner = () => h(centeredSpinner, { style: { position: 'fixed' } });
+  const authspinner = () => h(centeredSpinner, div({ style: { position: 'fixed' } }));
```

Comment:
Admittedly confusingly, the various spinners are implemented as render functions or elements instead of components and thus shouldn't be used with `h`.

```suggestion
  const authspinner = () => centeredSpinner({ style: { position: 'fixed' } });
```

Alternatively, `fixedSpinnerOverlay` (from src/components/common) might be more appropriate here.

```suggestion
  const authspinner = () => fixedSpinnerOverlay;
```